Sources/MapboxNavigation/WayNameView.swift

```diff
@@ -2,12 +2,161 @@ import Foundation
 import UIKit
 import Turf
 import MapboxMaps
+import MapboxDirections
 
 /**
  A label that is used to show a road name and a shield icon.
  */
 @objc(MBWayNameLabel)
-open class WayNameLabel: StylableLabel {}
+open class WayNameLabel: StylableLabel {
+    var spriteRepository = SpriteRepository.shared
```

Comment:
I did not have a chance to review `SpriteRepository` implementation it time, but I'd consider adding `Result` usage (which contains error and expected value) to completion handlers there, it'll give us the ability to track possible issues that might arise.

Reply:
I've also walked though APIs in `SpriteRepository` and think that some methods/properties should be simplified. For example: why do we need initializer with all default parameters? If it's not used anywhere and main point of access in `shared` property we can remove it. Some properties are public, I'd consider making them internal, as they're not meant to be used by anybody else for now.

Reply:
Thanks for the suggestion about the `Result` usage. Yes it should be good if we could use the `result` that has image/data/error for the completion handlers. But the issue is that when we update the `SpriteRepository` with the `ImageRepresentation` or the road shield. We may need to request the sprite info data, the sprite image and the single legacy shield icon at the same time. The three are different in types. 

And the result of the sprite info data would be parsed into dictionary and stored in cache, the sprite image and  the legacy shield icon would also be stored in cache in the `completion` of the downloading request of `imageDownloader`. So the `CompletionHandler` of the sprite update call in fact receives all the process. Right now I haven't found a way to do it separately. Because the single shield icon need both the Sprite info and Sprite image to conduct the cutting process.

Reply:
Yes, I'll remove the `shared` property and the initializer for the `SpriteRepository`, also change these public properties to internal because they're not used anywhere. For the default parameter, because the changes of the shield baseURL and map style don't happen at the same time, we need to stored the value when each one of them change.